src/pokemon.rs

```diff
@@ -1,7 +1,10 @@
+use num_enum::IntoPrimitive;
+
 use crate::define_enum_with_from_str;
 
 define_enum_with_from_str! {
-    #[derive(Clone, Debug, PartialEq)]
+    #[derive(Clone, Debug, PartialEq, IntoPrimitive)]
+    #[repr(u32)]
```

Comment:
```suggestion
    #[repr(u16)]
```

Curious if u32 was picked intentionally. 16 bits should be enough to identify a Pokemon for many more generations

Not that it makes a huge deal since this engine doesn't need to be optimized

Reply:
u32 wasn't picked intentionally, in fact I thought about changing it to a u16 at some point but forgot to change it.